libs/database/src/collab/disk_cache.rs

```diff
@@ -73,20 +116,46 @@ impl CollabDiskCache {
     } else if params.collab_type == CollabType::Document {
       tracing::info!("no embeddings to save for collab {}", params.object_id);
     }
+
     Ok(())
   }
 
   #[instrument(level = "trace", skip_all)]
   pub async fn get_collab_encoded_from_disk(
     &self,
+    workspace_id: &str,
     query: QueryCollab,
   ) -> Result<EncodedCollab, AppError> {
-    event!(
-      Level::DEBUG,
-      "try get {}:{} from disk",
-      query.collab_type,
-      query.object_id
-    );
+    tracing::debug!("try get {}:{} from s3", query.collab_type, query.object_id);
+    let key = collab_key(workspace_id, &query.object_id);
+    match self.s3.get_blob(&key).await {
+      Ok(resp) => {
+        let blob = resp.to_blob();
+        let now = Instant::now();
+        let decompressed = zstd::decode_all(&*blob)?;
```

Comment:
Just to make sure whether encode or decode should not stuck the tokio runtime.

Reply:
The problem with loading the decoding to separate thread is that it involves context switching between threads, which is kernel operation and each context switch takes the kernel about 5 μs (on average) to process. And this doesn't involve time required for creating a new thread (ie. if we use rayon). That means that we add ~10 μs for every decompression task.

I'd keep it as is, and if CPU indeed will be an issue, add rayon afterwards. We can also switch to lz4 (which is faster but has worse compression sizes) to reduce these times, since the thing is not merged yet.